sys/crypto/modes/ccm.c

```diff
@@ -105,8 +105,14 @@ int ccm_compute_adata_mac(cipher_t *cipher, const uint8_t *auth_data,
     if (auth_data_len > 0) {
         int len;
 
-        /* 16 octet block size + max. 10 len encoding  */
-        uint8_t auth_data_encoded[26], len_encoding = 0;
+        /* Create a block with the encoded length */
+        int block_size = cipher_get_block_size(cipher);
+        if (block_size > 16  || block_size < 0) {
+            DEBUG("UNSUPPORTED block size of the cipher: %d\n",
+                  block_size);
+            return -1;
```

Comment:
Is this check needed? AFAIK  CCM is only defined for use with 128-bit block ciphers. In which case I think an assert would be better, and it should be in the `cipher_decrypt_ccm` and `cipher_encrypt_ccm` functions IMO.

Reply:
But I think this would be unrelated to this PR, it could be in a separate commit though.

Reply:
Done: b46a018d49e048d7745fa7fd9e49645704ae1cb4